crypto/cmp/cmp_int.h

```diff
@@ -26,6 +26,130 @@
 # include <openssl/x509.h>
 # include <openssl/x509v3.h>
 
+/*
+ * this structure is used to store the context for CMP sessions
+ */
+struct ossl_cmp_ctx_st {
+    OSSL_cmp_log_cb_t log_cb; /* log callback for error/debug/etc. output */
+    OSSL_CMP_severity log_verbosity; /* level of verbosity of log output */
+
+    /* message transfer */
+    OSSL_cmp_transfer_cb_t transfer_cb; /* default: OSSL_CMP_MSG_http_perform */
+    void *transfer_cb_arg; /* allows to store optional argument to cb */
+    /* HTTP-based transfer */
+    char *serverPath;
+    char *serverName;
+    int serverPort;
+    char *proxyName;
+    int proxyPort;
+    int msgtimeout; /* max seconds to wait for each CMP message round trip */
+    int totaltimeout; /*
```

Comment:
nit: better move a long comment block like this, above the concerned field, it will ease readability, IMHO.

Reply:
I prefer to have the comments just next to the field decls, which makes it clear to which field they belong.
Unfortunately this does not work very well with longer comments.